src/borg/remote.py

```diff
@@ -646,8 +645,8 @@ def borg_cmd(self, args, testing):
             except AttributeError:
                 pass
         env_vars = []
-        if yes(env_var_override='BORG_HOSTNAME_IS_UNIQUE', env_msg=None, prompt=False):
-            env_vars.append('BORG_HOSTNAME_IS_UNIQUE=yes')
+        if not hostname_is_unique():
+            env_vars.append('BORG_HOSTNAME_IS_UNIQUE=no')
```

Comment:
there is the remaining problem that passing this is not effective with a forced command (see your other PR, you just throw away the env vars).